src/pages/AllSpecies/AllSpecies.tsx

```diff
@@ -63,12 +64,39 @@ const AllSpecies = ({ characters }: AllSpeciesProps) => {
     })
   }
 
+  useEffect(() => {
+    if (
+      charactersState.sortFactor ||
+      charactersState.homeworldFilterTarget ||
+      charactersState.genderFilterTarget
+    ) {
+      setShowClearButton(true)
+    } else {
+      setShowClearButton(false)
+    }
+  }, [
+    charactersState.sortFactor,
+    charactersState.homeworldFilterTarget,
+    charactersState.genderFilterTarget,
+  ])
+
```

Comment:
This is quite an aggressive checking you are doing 😄 I'd make a variable out of this if condition to make is a bit more readable;

```suggestion
  const shouldClearButtonPresent = charactersState.sortFactor ||
      charactersState.homeworldFilterTarget ||
      charactersState.genderFilterTarget
      
  useEffect(() => {
      setShowClearButton(shouldClearButtonPresent)
  }, [shouldClearButtonPresent])
```